analyses/snv-callers/util/plot_functions.R

```diff
@@ -0,0 +1,274 @@
+    ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
+    ggplot2::xlab("") +
+    ggplot2::ylab("Count") +
+    colorblindr::scale_fill_OkabeIto()
+
+  return(barplot)
+}
+
+depth_vs_vaf_plot <- function(vaf_df, exp_strategy = "BOTH") {
+  # Plot read depth versus variant allele fraction. If both experimental strategies
+  # are included, the plot is made color-coded by strategy.
+  #
+  # Args:
+  #   vaf_df: MAF formatted data that has been turned into a data.frame and has
+  #           been run through `set_up_maf` with metadata (specifically the
+  #           `experimental_strategy column added.
+  #   exp_strategy: argument to specify whether to plot `wgs`, `wxs` samples or
+  #   `both` Case insensitive.
+  #
+  # Make this argument case insensitive
+  exp_strategy <- toupper(exp_strategy)
+
+  # Filter out based on experimental stategy if specified
+  if (exp_strategy != "BOTH") {
+    vaf_df <- vaf_df %>%
+      dplyr::filter(experimental_strategy == exp_strategy)
+  }
+
+  # Plot this as a scatterplot
+  barplot <- ggplot2::ggplot(
+    vaf_df,
+    ggplot2::aes(x = log2(n_depth), y = vaf)
+  ) +
+    ggplot2::theme_classic()
+
+  # Get rid of legend if both data aren't being plotted
+  if (exp_strategy != "BOTH") {
+    barplot <- barplot +
+      ggplot2::geom_point(color = "navyblue", alpha = 0.15) +
+      ggplot2::theme(legend.position = "none")
+  } else {
+    barplot <- barplot +
+      ggplot2::geom_point(ggplot2::aes(color = experimental_strategy), alpha = 0.15) +
+      colorblindr::scale_color_OkabeIto()
+  }
+  return(barplot)
+}
+
+snv_region_plot <- function(maf_annot, exp_strategy = "BOTH", filter_cutoff = 0) {
+  # Plot the genomic region analysis from the annotation added from `annotatr_maf`
+  # as a barplot with the number of mutations found in each type of genomic region.
+  #
+  # Args:
+  #   maf_annot: MAF formatted data that has been turned into a data.frame and has
+  #           been run through `annotr_maf`
+  #   exp_strategy: argument to specify whether to plot `wgs`, `wxs` samples or
+  #   `both`.
+  #   filter_cutoff: A numeric number to only keep groups larger than it.
+  #
+  # Returns:
+  # A barplot with the number of mutations that are found within each type of
+  # genomic region in the `type` column
+  #
+  # Make this argument case insensitive
+  exp_strategy <- toupper(exp_strategy)
+
+  # Filter out based on experimental stategy if specified
+  if (exp_strategy != "BOTH") {
+    maf_annot <- maf_annot %>%
+      dplyr::filter(experimental_strategy == exp_strategy)
+  }
+
+  # Count the number of each type of base change
+  type_count_df <- maf_annot %>%
+    dplyr::group_by(type, experimental_strategy) %>%
+    dplyr::summarise(type_count = dplyr::n()) %>%
+    dplyr::filter(type_count > filter_cutoff)
+
+  # Plot this as a barplot
+  barplot <- ggplot2::ggplot(
+    type_count_df,
+    ggplot2::aes(x = reorder(type, -type_count), y = type_count)
```

Comment:
Do we want to reorder by count here, or have a consistent order? Consistency will make it easier to compare between plots with different data sets. Note that in the consistent order case, if for some amazing reason there were none of a particular mutation, we _would_ still want to include that zero bar to preserve order.

Reply:
I'll make it a consistent order.